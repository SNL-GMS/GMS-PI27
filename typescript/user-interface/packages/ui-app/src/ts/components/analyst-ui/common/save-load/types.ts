import type { AppState } from '@gms/ui-state';

export interface GmsExport {
  versionInfo: string;
  reduxStore: AppState;
  waveformStore: Record<string, Float64Array>;
}
