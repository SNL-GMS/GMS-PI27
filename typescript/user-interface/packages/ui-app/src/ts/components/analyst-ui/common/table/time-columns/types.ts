import type { Row } from '@gms/ui-core-components';
import type { ArrivalTime } from '@gms/ui-state';

export interface TimeRow extends Row {
  time: ArrivalTime;
}
