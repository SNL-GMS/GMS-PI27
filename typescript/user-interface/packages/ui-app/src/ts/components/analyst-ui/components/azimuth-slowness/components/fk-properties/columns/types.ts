import type { CellRendererParams, Row, ValueFormatterParams } from '@gms/ui-core-components';
import type { ICellRendererParams, IHeaderParams } from 'ag-grid-community';
import Immutable from 'immutable';

/**
 * Column names for the FK Properties table
 */
export enum FkPropertiesColumn {
  description = 'description',
  peak = 'peak',
  predicted = 'predicted',
  measured = 'measured',
  residual = 'residual'
}

/**
 * Used to match the display strings to values in the FK Properties table
 */
export const FkPropertiesColumnDisplayStrings: Immutable.Map<FkPropertiesColumn, string> =
  Immutable.Map<FkPropertiesColumn, string>([
    [FkPropertiesColumn.description, ''],
    [FkPropertiesColumn.peak, 'Peak'],
    [FkPropertiesColumn.predicted, 'Predicted'],
    [FkPropertiesColumn.measured, 'Measured'],
    [FkPropertiesColumn.residual, 'Residual']
  ]);

/** Interface that defines a cell of data */
export interface DataCell {
  value: number;
  uncertainty: number | undefined;
}

/** Interface for defining a row in the properties table */
export interface FkPropertiesRow extends Row {
  id: string;
  description: string;
  phaseIsInvalid: boolean;
  eventIsOpen: boolean;
  hasAnalystMeasurement?: boolean;
  extrapolated: boolean;
  peak: DataCell | undefined;
  predicted: DataCell | undefined;
  measured: DataCell | undefined;
  residual: DataCell | undefined;
}

export type FkPropertiesCellValueFormatterParams = ValueFormatterParams<
  FkPropertiesRow,
  unknown,
  DataCell,
  CellRendererParams<FkPropertiesRow, unknown, any, ICellRendererParams, IHeaderParams>,
  IHeaderParams
>;
