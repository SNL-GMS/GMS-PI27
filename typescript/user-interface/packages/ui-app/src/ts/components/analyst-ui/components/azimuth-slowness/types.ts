import type { FkTypes, SignalDetectionTypes } from '@gms/common-model';
import type GoldenLayout from '@gms/golden-layout';

/**
 * Used to return a super set of the fk configuration from the fk config popover
 */
export interface FkConfigurationWithUnits {
  fkUnitToDisplay: FkTypes.FkUnits;
}

/**
 * Props for the top-level Azimuth Slowness Container
 */
export interface AzimuthSlownessProps {
  /** passed in from golden-layout */
  glContainer?: GoldenLayout.Container;
}

export interface SubscriptionAction {
  (
    list: SignalDetectionTypes.SignalDetection[],
    index: number,
    prev: SignalDetectionTypes.SignalDetection[],
    currentIteree: SignalDetectionTypes.SignalDetection
  ): void;
}

/**
 * State of the AzimuthSlownessPanel
 */
export interface AzimuthSlownessPanelState {
  currentMovieSpectrumIndex: number;
  selectedFkUnit: FkTypes.FkUnits;
  phaseMenuVisibility: boolean;
  draggingDivider: boolean;
  dividerStartX: number;
  minDividerX: number;
  maxDividerX: number;
}
