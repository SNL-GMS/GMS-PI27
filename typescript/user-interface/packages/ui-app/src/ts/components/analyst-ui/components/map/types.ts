import type { SignalDetectionTypes } from '@gms/common-model';
import type * as Cesium from 'cesium';
import type { CesiumMovementEvent } from 'resium';

import type { EventFilterOptions, EventRow } from '../events/types';

export type MapRightClickHandler = (
  movement: CesiumMovementEvent,
  target: Cesium.Entity,
  latitude: number,
  longitude: number
) => void;

/**
 * Event source for map display
 * Extending EventRow type with additional geoOverlappingEvents property
 */
export interface MapEventSource extends EventRow {
  geoOverlappingEvents?: number;
}

/**
 * Values to be added to signal detection entity property bag
 */
export interface MapSDEntityPropertyBagDefinitions {
  readonly id: string;
  readonly type: string;
  readonly detectionTime: {
    detectionTimeValue: string;
    detectionTimeUncertainty: number;
  };
  readonly azimuth: {
    azimuthValue: number;
    azimuthUncertainty: number;
  };
  readonly slowness: {
    slownessValue: number;
    slownessUncertainty: number;
  };
  readonly phaseValue: SignalDetectionTypes.PhaseTypeMeasurementValue;
  readonly deleted: boolean;
  readonly associated: boolean;
  readonly associatedEventTimeValue: string;
  readonly signalDetectionBaseColor: string;
  readonly status: SignalDetectionTypes.SignalDetectionStatus;
  readonly edgeSDType: EventFilterOptions;
  readonly stationName: string;
  readonly isSelectedOrActionTarget: boolean;
}

export enum UncertaintyEllipse {
  CONFIDENCE = 'CONFIDENCE',
  COVERAGE = 'COVERAGE'
}
