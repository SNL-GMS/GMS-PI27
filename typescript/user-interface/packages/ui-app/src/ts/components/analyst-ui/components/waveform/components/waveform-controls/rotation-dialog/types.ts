import type { ChannelTypes, SignalDetectionTypes, StationTypes } from '@gms/common-model';
import type { ValidationDefinition } from '@gms/ui-core-components';
import type { Message } from '@gms/ui-core-components/lib/components/ui-widgets/form/types';
import { createNonPayloadAction, createPayloadAction } from '@gms/ui-state';

type StationName = string;
type ChannelName = string;
export type DefaultRotatedChannelsRecord = Record<StationName, [ChannelName, ChannelName]>;

export interface RotationErrorMessages {
  latInvalidMessage: Message;
  lonInvalidMessage: Message;
  leadInvalidMessage: Message;
  durationInvalidMessage: Message;
  channelInvalidMessage: Message;
  azimuthInvalidMessage: Message;
  signalDetectionInvalidMessage: Message;
  infoMessage: Message;
  stationInvalidMessage: Message;
}

export interface RotationErrorAction {
  type: keyof RotationErrorMessages;
  payload: Message;
}

export type RotationMessageAction = RotationErrorAction | { type: 'clearMessages' };

export type InputMode = 'signal-detection-mode' | 'station-phase-mode' | null;
export type RotationInterpolationMethod = string;
export type SteeringMode = 'reference-location' | 'azimuth' | 'measured-azimuth';
export type LeadDurationMode = 'default-station-phase' | 'custom-lead-duration';

export interface StationPhaseConfig {
  channelSampleRateTolerance: number;
  channelOrientationTolerance: number;
  locationToleranceKm: number;
}
export interface RotationDialogState {
  // easily identify a rotation dialog state object
  isRotationDialogState: true;

  /**
   * Azimuth state. Values should be strings to allow decimals, per Blueprint documentation
   * These should be validated and converted into numbers before use.
   */
  azimuth: string;

  /**
   * The mode for choosing rotation inputsâ€”signal detection mode or station/phase mode
   */
  inputMode: InputMode;

  /**
   * The selected interpolation method
   */
  interpolation: RotationInterpolationMethod;

  /**
   * Latitude state. Values should be strings to allow decimals, per Blueprint documentation
   * These should be validated and converted into numbers before use.
   */
  latitude: string;

  /**
   * The mode for choosing the lead and duration for the rotated waveforms, either configured by
   * station/phase, or using custom values.
   */
  leadDurationMode: LeadDurationMode;

  /**
   * The amount of time before the reference time at which to start generating rotated waveforms.
   * Values should be strings to allow decimals, per Blueprint documentation.
   * These should be validated and converted into numbers before use.
   */
  leadSecs: string;

  /**
   * The duration of the rotated waveforms.
   * Values should be strings to allow decimals, per Blueprint documentation.
   * These should be validated and converted into numbers before use.
   */
  durationSecs: string;

  /**
   * Longitude state. Values should be strings to allow decimals, per Blueprint documentation.
   * These should be validated and converted into numbers before use.
   */
  longitude: string;

  /**
   * The rotation phase based on the user's selected phase. Default is taken from rotation configuration
   * for the currently open activity
   */
  rotationPhase: string | undefined;

  /**
   * Steering mode. This controls how we get an azimuth for each signal detection.
   */
  steeringMode: SteeringMode;

  /**
   * The Channels to use for rotation.
   * The user may override these
   */
  targetChannels: ChannelTypes.Channel[];

  /**
   * The signal detections to use for rotation.
   * Station, default channels, phase, and azimuth will be used from these signal detections
   * for each rotated waveform.
   * Only used in signal-detection-mode
   */
  targetSignalDetections: SignalDetectionTypes.SignalDetection[];

  /**
   * The stations to use for rotation.
   * The user may override these
   */
  targetStations: StationTypes.Station[];

  /**
   * The Channels that may be chosen in the channel selector in the rotation dialog.
   */
  validChannels: ChannelTypes.Channel[];

  /**
   * The set of signal detections that may be chosen in the selector.
   * Only used in signal-detection-mode
   */
  validSignalDetections: SignalDetectionTypes.SignalDetection[];

  /**
   * The stations that may be chosen in the station selector in the rotation dialog.
   */
  validStations: StationTypes.Station[];

  /**
   * True if the rotation templates for rotation targets have been fetched
   */
  hasRotationTemplates: boolean;
}

/**
 * The actions that may be dispatched to change the rotation dialog state
 */
export const rotationDialogActions = {
  setAzimuth: createPayloadAction<string>()('setAzimuth'),
  setDuration: createPayloadAction<string>()('setDuration'),
  setInputMode: createPayloadAction<InputMode>()('setInputMode'),
  setInterpolation: createPayloadAction<string>()('setInterpolation'),
  setLatitude: createPayloadAction<string>()('setLatitude'),
  setLongitude: createPayloadAction<string>()('setLongitude'),
  setLead: createPayloadAction<string>()('setLead'),
  setLeadDurationMode: createPayloadAction<LeadDurationMode>()('setLeadDurationMode'),
  setSteeringMode: createPayloadAction<SteeringMode>()('setSteeringMode'),
  setPhase: createPayloadAction<string>()('setPhase'),
  setTargetSignalDetections: createPayloadAction<SignalDetectionTypes.SignalDetection[]>()(
    'setTargetSignalDetections'
  ),
  setTargetStations: createPayloadAction<StationTypes.Station[]>()('setTargetStations'),
  setTargetChannels: createPayloadAction<string[]>()('setTargetChannels')
} as const;

// Extract a list of action type stings to use in type guards
const rotationDialogActionTypeStrings = Object.values(rotationDialogActions).map(
  actionBuilder => actionBuilder({} as never).type
);

export type RotationDialogActionType = keyof RotationDialogState;

/**
 * Type guard which determines if an object is a {@link RotationAction}
 *
 * @param action any object which may or may not be a {@link RotationAction}
 * @returns true if the action is a {@link RotationAction}
 */
export function isRotationAction(action: {
  type: string;
  payload: unknown;
}): action is RotationAction {
  return rotationDialogActionTypeStrings.includes((action as any).type);
}

/**
 * The types of the actions created by each of the {@link rotationDialogActions}
 */
export type RotationAction = ReturnType<
  (typeof rotationDialogActions)[keyof typeof rotationDialogActions]
>;

const createMessagePayloadAction = createPayloadAction<Message>();
const createMessageNonPayloadAction = createNonPayloadAction();

/**
 * Action generators for rotation error message actions which may be used\
 * to dispatch error messages.
 *
 * @example ```
 * dispatchErrorMessage(
 *  rotationErrorActions.setAzimuthInvalidMessage(
 *    {summary: "Invalid azimuth", intent: 'danger'}
 * ));
 * ```
 */
export const rotationErrorActions = {
  setAzimuthInvalidMessage: createMessagePayloadAction('azimuthInvalidMessage'),
  setChannelInvalidMessage: createMessagePayloadAction('channelInvalidMessage'),
  setDurationInvalidMessage: createMessagePayloadAction('durationInvalidMessage'),
  setLatInvalidMessage: createMessagePayloadAction('latInvalidMessage'),
  setLeadInvalidMessage: createMessagePayloadAction('leadInvalidMessage'),
  setLonInvalidMessage: createMessagePayloadAction('lonInvalidMessage'),
  setSignalDetectionInvalidMessage: createMessagePayloadAction('signalDetectionInvalidMessage'),
  setInfoMessage: createMessagePayloadAction('infoMessage'),
  clearMessages: createMessageNonPayloadAction('clearMessages')
};

// Extract a list of error action type stings to use in type guards
export const rotationErrorActionTypeStrings = Object.values(rotationErrorActions).map(
  // cast to never because we do not care about the payload, we just want to get the type strings
  actionBuilder => actionBuilder({} as never).type
);

export type RotationErrorTypes = keyof RotationErrorMessages;

export interface RotationDialogMessage extends Message {
  errorType: RotationErrorTypes;
}

export type RotationValidationDefinition = ValidationDefinition<
  RotationDialogState,
  RotationDialogMessage | ((state: RotationDialogState) => RotationDialogMessage)
>;

/**
 * Type guard which determines if an object is a {@link RotationErrorAction}
 *
 * @param action any object which may or may not be a {@link RotationErrorAction}
 * @returns true if the action is a {@link RotationErrorAction}
 */
export function isRotationErrorAction(action: {
  type: string;
  payload?: unknown;
}): action is RotationErrorAction {
  return rotationErrorActionTypeStrings.includes((action as any).type);
}
