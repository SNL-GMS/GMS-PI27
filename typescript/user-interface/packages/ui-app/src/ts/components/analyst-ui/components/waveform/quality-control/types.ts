import type { Row } from '@gms/ui-core-components';

/**
 * Interface that describes the QC Mask history information.
 */
export interface QcMaskHistoryRow extends Row {
  author: string;
  category: string;
  type: string;
  startTime: number;
  channelName: string[];
  endTime: number;
  rationale: string;
  effectiveAt: number;
  rejected: string;
}
