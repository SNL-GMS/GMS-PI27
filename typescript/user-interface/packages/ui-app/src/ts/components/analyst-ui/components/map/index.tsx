import { IanDisplays, toDisplayTitle } from '@gms/common-model/lib/displays/types';
import { nonIdealStateWithSpinner } from '@gms/ui-core-components';
import React from 'react';

import type { MapComponentProps } from './map-component';

const IANMapLazy = React.lazy(async () =>
  import(/* webpackChunkName: 'ui-app-ian-map' */ './map-component').then(module => ({
    default: module.IANMap
  }))
);

export function IANMap(props: MapComponentProps) {
  return (
    <React.Suspense
      fallback={nonIdealStateWithSpinner(toDisplayTitle(IanDisplays.MAP), 'Loading Display')}
    >
      {/* eslint-disable-next-line react/jsx-props-no-spreading */}
      <IANMapLazy {...props} />
    </React.Suspense>
  );
}
