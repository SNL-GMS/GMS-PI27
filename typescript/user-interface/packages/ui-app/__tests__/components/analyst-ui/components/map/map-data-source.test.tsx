import { SignalDetectionTypes } from '@gms/common-model';
import {
  eventData,
  processingAnalystConfigurationData,
  signalDetectionsData
} from '@gms/common-model/__tests__/__data__';
import type { Entity } from 'cesium';
import cloneDeep from 'lodash/cloneDeep';

import { createSignalDetectionEntities } from '~analyst-ui/components/map/create-ian-entities';

import { EventFilterOptions } from '../../../../../src/ts/components/analyst-ui/components/events/types';
import {
  sdOnMouseEnterHandler,
  sdOnMouseLeaveHandler
} from '../../../../../src/ts/components/analyst-ui/components/map/map-utils';

describe('map data source', () => {
  const associatedEvent = cloneDeep(eventData);
  signalDetectionsData.forEach(sd => {
    associatedEvent.eventHypotheses[0].associatedSignalDetectionHypotheses.push(
      sd.signalDetectionHypotheses[0]
    );
  });

  const entities: Entity = createSignalDetectionEntities(
    {
      signalDetections: signalDetectionsData,
      selectedSdIds: [],
      edgeSDType: EventFilterOptions.INTERVAL,
      status: SignalDetectionTypes.SignalDetectionStatus.OTHER_ASSOCIATED,
      sdVisibility: true,
      signalDetectionLengthMeters: null
    },
    [],
    [],
    processingAnalystConfigurationData.uiThemes[0],
    []
  )[0];

  const eventTarget = { id: entities };

  test('sdOnMouseEnterHandler', () => {
    const result = sdOnMouseEnterHandler({} as any, eventTarget as any);
    expect(result).toMatchSnapshot();
  });

  test('sdOnMouseLeaveHandler', () => {
    const result = sdOnMouseLeaveHandler({} as any, eventTarget as any);
    expect(result).toMatchSnapshot();
  });
});
