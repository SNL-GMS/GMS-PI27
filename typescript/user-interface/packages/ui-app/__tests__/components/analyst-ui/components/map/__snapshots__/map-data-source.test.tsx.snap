// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map data source sdOnMouseEnterHandler 1`] = `undefined`;

exports[`map data source sdOnMouseLeaveHandler 1`] = `undefined`;
